src/util.h

```diff
@@ -242,6 +245,15 @@ class ArgsManager
      */
     bool IsArgSet(const std::string& strArg) const;
 
+    /**
+     * Return true if the argument was originally passed as a negated option,
+     * i.e. -nofoo.
+     *
+     * @param strArg Argument to get (e.g. "-foo")
+     * @return true if the argument was passed negated
+     */
+    bool IsArgNegated(const std::string& strArg) const;
```

Comment:
I don't think we should expose this in the code base. Historically, `-nofoo` always implied `-foo=0`, so use cases within the code  base should always compare with the string `"0"` to check if foo was unset.

Yes, I know that that would forbid use cases, such as `-myfile=0`, but imo that is acceptable and desirable to keep previous behavior.

Reply:
> Historically, -nofoo always implied -foo=0, so use cases within the code base should always compare with the string "0" to check if foo was unset.

This is true for boolean options, and should still be true with this PR. I don't think it was ever historically true that specifying "0" or "false" for path type options would disable those options. At least this isn't the case currently for `-conf` `-debuglogfile` `-pid` `-rpccookiefile` `-wallet` or other path options. So whether or not special path strings are a good idea (I think they are a bad idea), I don't think there is a historic argument for having them.

Reply:
I wanted to raise the point that we specifically check for the string "0" in some non-bool arguments to see if they are disabled. E.g. `-connect=0` or `-debug=0`. So in case someone were to use `IsArgNegated` for those, it would break it.

Again, only bike-shedding here, so please ignore.

Reply:
I do kind of agree with @ryanofsky that not being able to create a log file named "0" (or "false") is weird. Tracking negation makes the arguments more strongly typed in some kind of weird sense. I also felt like it made my printtoconsole change a little easier to understand, because it removes the code that calls `GetBoolArg()` on a non-boolean field.